include/openmc/plot.h

```diff
@@ -192,61 +235,127 @@ T PlotBase::get_map() const
-  void set_user_colors(pugi::xml_node plot_node);
   void set_meshlines(pugi::xml_node plot_node);
-  void set_mask(pugi::xml_node plot_node);
-  void set_overlap_color(pugi::xml_node plot_node);
 
-  // Members
 public:
-  int id_;                        //!< Plot ID
+  // Add mesh lines to ImageData
+  void draw_mesh_lines(ImageData& data) const;
+  void create_image() const;
+  void create_voxel() const;
+
+  virtual void create_output() const;
+  virtual void print_info() const;
+
   PlotType type_;                 //!< Plot type (Slice/Voxel)
-  PlotColorBy color_by_;          //!< Plot coloring (cell/material)
   int meshlines_width_;           //!< Width of lines added to the plot
   int index_meshlines_mesh_ {-1}; //!< Index of the mesh to draw on the plot
   RGBColor meshlines_color_;      //!< Color of meshlines on the plot
-  RGBColor not_found_ {WHITE};    //!< Plot background color
-  RGBColor overlap_color_ {RED};  //!< Plot overlap color
-  vector<RGBColor> colors_;       //!< Plot colors
-  std::string path_plot_;         //!< Plot output filename
+};
+
+class ProjectionPlot : public PlottableInterface {
+
+public:
+  ProjectionPlot(pugi::xml_node plot);
+
+  virtual void create_output() const;
+  virtual void print_info() const;
+
+private:
+  void set_output_path(pugi::xml_node plot_node);
+  void set_look_at(pugi::xml_node node);
+  void set_camera_position(pugi::xml_node node);
+  void set_field_of_view(pugi::xml_node node);
+  void set_pixels(pugi::xml_node node);
+  void set_opacities(pugi::xml_node node);
+  void set_orthographic_width(pugi::xml_node node);
+  void set_wireframe_thickness(pugi::xml_node node);
+  void set_wireframe_ids(pugi::xml_node node);
+  void set_wireframe_color(pugi::xml_node node);
+
+  /* Used for drawing wireframe and colors. We record the list of
+   * surface/cell/material intersections and the corresponding lengths as a ray
+   * traverses the geometry, then color by iterating in reverse.
+   */
+  struct TrackSegment {
+    int id;        // material or cell ID (which is being colored)
+    double length; // length of this track intersection
+
+    /* Recording this allows us to draw edges on the wireframe. For instance
+     * if two surfaces bound a single cell, it allows drawing that sharp edge
+     * where the surfaces intersect.
+     */
+    int surface; // last surface ID intersected in this segment
+    TrackSegment(int id_a, double length_a, int surface_a)
+      : id(id_a), length(length_a), surface(surface_a)
+    {}
+  };
+
+  std::array<int, 2> pixels_;       // pixel dimension of resulting image
+  double horizontal_field_of_view_ {70.0}; // horiz. f.o.v. in degrees
+  Position camera_position_;        // where camera is
+  Position look_at_;                // point camera is centered looking at
+  Direction up_ {0.0, 0.0, 1.0};    // which way is up
+  std::vector<int>
+    wireframe_ids_; // which color IDs should be wireframed. If empty, all
```

Comment:
Looks like the end of this comment got cut off accidentally

Reply:
fixed!